src/Physics/Common/QvalueShifter.cxx

```diff
@@ -0,0 +1,93 @@
+//_________________________________________________________________________
+/*
+ Copyright (c) 2003-2020, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+ Original code contributed by J.Tena and M.Roda
+ Substantial code refactorizations by the core GENIE group.
+*/
+//_________________________________________________________________________
+
+#include "Framework/Algorithm/AlgConfigPool.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Physics/Common/QvalueShifter.h"
+#include "Framework/Utils/StringUtils.h" 
+
+using namespace genie;
+
+//_________________________________________________________________________
+QvalueShifter::QvalueShifter() : 
+  Algorithm("genie::QvalueShifter") 
+{
+  
+}
+//_________________________________________________________________________
+QvalueShifter::QvalueShifter(string config) : 
+  Algorithm("genie::QvalueShifter",config) 
+{
+  
+}
+//_________________________________________________________________________
+QvalueShifter::~QvalueShifter()
+{
+
+}
+//_________________________________________________________________________
+void QvalueShifter::Configure(const Registry & config)
+{
+    Algorithm::Configure(config);
+    this->LoadConfig();
+}
+//____________________________________________________________________________
+void QvalueShifter::Configure(string config)
+{
+    Algorithm::Configure(config);
+    this->LoadConfig();
+}
+//_________________________________________________________________________
+
+  double QvalueShifter::Shift( const Target & target ) const {
+    // Get Target pdg
+    int pdg_target = target.Pdg() ;
+
+    // Find Pdg in map:
+    if ( fRelShift.find(pdg_target) != fRelShift.end() ) {
```

Comment:
Write it more efficiently.
You loop over the map twice with your code: first to check if the pdg is there, second to return the value.
Just save the returned value in the first place:
```C++
const auto it = fRelShift.find(pdg_target) ;
if ( it != fRelShift.end() ) {
  return it -> second ;
}
```